src/EFCore.PG/Migrations/NpgsqlMigrationsSqlGenerator.cs

```diff
@@ -43,6 +43,19 @@ public NpgsqlMigrationsSqlGenerator([NotNull] MigrationsSqlGeneratorDependencies
         {
         }
 
+        // Hack - Discuss this!!
+        // Issue - https://github.com/npgsql/Npgsql.EntityFrameworkCore.PostgreSQL/issues/268
+        // This is horrible, but we'll stay until we solve it!
+        private static string TruncateBigName(string name)
+        {
+            if(name?.Length>64)
+            {
+                name = name.Substring(0, 32) + Guid.NewGuid().ToString("N");
+            }
+
+            return name;
```

Comment:
@roji 
I know this is awful, we need to for now, not to break many tests!

I wonder how to handle this better later.

Reply:
@roji ,
Let's look at this basic example [here](https://github.com/aspnet/EntityFrameworkCore/blob/dev/test/EFCore.Sqlite.FunctionalTests/UpdatesSqliteTest.cs#L22):

If I have this:
```
LoginEntityTypeWithAnExtremelyLongAndOverlyConvolutedNameThatIsUsedToVerifyThatTheStoreIdentifierGenerationLengthLimitIsWorkingCorrectly
LoginEntityTypeWithAnExtremelyLongAndOverlyConvolutedNameThatIsUsedToVerifyThatTheStoreIdentifierGenerationLengthLimitIsWorkingBad
```
I will have this, duplicated:
```
LoginEntityTypeWithAnExtremelyLongAndOverlyConvolutedNameThatIs~
LoginEntityTypeWithAnExtremelyLongAndOverlyConvolutedNameThatIs~
```

So as not to break the tests, I put this hack, as I said, it will be temporary.
That's just for a more basic example, there are others too!

I believe we can keep this for now.

Reply:
Sure, but your hack with the GUID doesn't exist in the SQL Server or Sqlite provider... What test breaks if we remove it? And when you say it's temporary, when do you plan to remove it?

Reply:
Yes, I plan to remove it, sorry for my English.

Some tests were being broken, so I did it, so I'm looking at other things and I'll come back to that later!

Reply:
I can not remember what they were now, in 40 minutes I'll look at them!

Reply:
No problem. In general I prefer that we leave tests broken rather than commit temporary hacks to make them pass... A failing test will continue failing, but it's easy to forget a temporary hack when releasing...